### PR TITLE
Created a component for test result history list, test result history…

### PR DESCRIPTION
**Pull request by VikkyMaru**
… and test results. Implemented a simple method for getting test result history by id.

### CODE REVIEW

File: Inception/ClientApp/src/App/Home/LinkTestResultComponent.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        selector: "LinkTestResult",
+        styleUrls: ["LinkTestResultComponent.css"],
+        templateUrl: "LinkTestResultComponent.html"
+    }
+    )
+export class LinkTestResultComponent implements OnInit
+{
+    @Input()
+    public linkTestResult: LinkTestResult;
```
Comment (PavelLev) [inline Inception/ClientApp/src/App/Home/LinkTestResultComponent.ts]: First letter should be in upper case

File: Inception/ClientApp/src/App/AppModule.ts [hunk 2]
```diff
@@ -1,30 +1,41 @@
...
 
+import { AppRouting } from "./AppRouting";
+import { AlertModule, ButtonsModule } from "ngx-bootstrap";
+import { TestResultHistoryListComponent } from "./Home/TestResultHistoryListComponent";
+import { Resolver } from "./Resolver";
+import { Service } from "./Services";
+import { TestResultHistoryComponent } from "./Home/TestResultHistoryComponent";
+import { LinkTestResultComponent } from "./Home/LinkTestResultComponent";
 
 @NgModule({
```
Comment (PavelLev) [inline Inception/ClientApp/src/App/AppModule.ts]: "(" and "{" should be on separate lines with single indent

File: Inception/ClientApp/src/App/AppRouting.ts [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
 import { NgModule } from "@angular/core";
+import { TestResultHistoryComponent } from "./Home/TestResultHistoryComponent";
 
 const Routes: Routes = [
-    { path: '', component: HomeComponent, pathMatch: 'full' },
+    { path: ":id", component: TestResultHistoryComponent },
+    { path: "", component: HomeComponent, pathMatch: "full" }
 ];
 
 @NgModule({
```
Comment (PavelLev) [inline Inception/ClientApp/src/App/AppRouting.ts]: "(" and "{" should be on separate lines with single indent

File: Inception/ClientApp/src/App/Home/HomeComponent.ts [hunk 4]
```diff
@@ -1,15 +1,29 @@
 import { OnInit, Component } from "@angular/core";
+import { Service } from "../Services";
+import { SiteTestResult } from "../Models";
 
 @Component(
```
Comment (PavelLev) [inline Inception/ClientApp/src/App/Home/HomeComponent.ts]: "(" and ")" should on separate lines with single indent

File: Inception/ClientApp/src/App/Home/TestResultHistoryComponent.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    public SiteTestResult: SiteTestResult;
+
+    constructor(private service: Service, private route: ActivatedRoute)
+    {
+
+    }
+
+    public ngOnInit()
+    {
+        this.SiteTestResult = this.service.getSiteTestResult(this.route.snapshot.params["id"]);
```
Comment (PavelLev) [inline Inception/ClientApp/src/App/Home/TestResultHistoryComponent.ts]: First letter of method should be in upper case

File: Inception/ClientApp/src/App/Home/TestResultHistoryListComponent.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    public SiteTestResult: SiteTestResult;
+
+    constructor(private service: Service, private route: ActivatedRoute)
+    { }
+
+
+
+    public ngOnInit()
+    {
+       // this.siteTestResult = this.route.snapshot.data["siteTestResult"];
```
Comment (PavelLev) [inline Inception/ClientApp/src/App/Home/TestResultHistoryListComponent.ts]: Remove junk code

File: Inception/ClientApp/src/App/Models.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+export class SiteTestResult
```
Comment (PavelLev) [inline Inception/ClientApp/src/App/Models.ts]: Extract in separate files with corresponding names

File: Inception/ClientApp/src/App/Services.ts [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+            {
+                subject.next(LinkTestResults);
+                subject.complete();
+            },
+            100);
+
+        return subject;
+    }
+
+    public getSiteTestResult(id: string): SiteTestResult
```
Comment (PavelLev) [inline Inception/ClientApp/src/App/Services.ts]: First letter of method should be in upper case

File: Inception/ClientApp/src/App/Services.ts [hunk 9]
```diff
@@ -0,0 +1,123 @@
+import { LinkTestResult, SiteTestResult } from "./Models";
+import { Observable } from "rxjs/Observable";
+import { Subject } from "rxjs/Subject";
+import { Injectable } from "@angular/core";
+
+@Injectable()
+export class Service {
```
Comment (PavelLev) [inline Inception/ClientApp/src/App/Services.ts]: Each service should be in separate file (and desirable have descriptive name)

File: Inception/ClientApp/src/App/Home/LinkTestResultComponent.ts (hunk 1 above)
Comment (VikkyMaru) [reply Inception/ClientApp/src/App/Home/LinkTestResultComponent.ts]: +

Comment (VikkyMaru) [reply Inception/ClientApp/src/App/Home/LinkTestResultComponent.ts]: +

File: Inception/ClientApp/src/App/AppModule.ts (hunk 2 above)
Comment (VikkyMaru) [reply Inception/ClientApp/src/App/AppModule.ts]: +

File: Inception/ClientApp/src/App/AppRouting.ts (hunk 3 above)
Comment (VikkyMaru) [reply Inception/ClientApp/src/App/AppRouting.ts]: +

File: Inception/ClientApp/src/App/Home/HomeComponent.ts (hunk 4 above)
Comment (VikkyMaru) [reply Inception/ClientApp/src/App/Home/HomeComponent.ts]: +

File: Inception/ClientApp/src/App/Home/TestResultHistoryListComponent.ts (hunk 6 above)
Comment (VikkyMaru) [reply Inception/ClientApp/src/App/Home/TestResultHistoryListComponent.ts]: +

File: Inception/ClientApp/src/App/Models.ts (hunk 7 above)
Comment (VikkyMaru) [reply Inception/ClientApp/src/App/Models.ts]: +

File: Inception/ClientApp/src/App/Services.ts (hunk 8 above)
Comment (VikkyMaru) [reply Inception/ClientApp/src/App/Services.ts]: +

File: Inception/ClientApp/src/App/Services.ts (hunk 9 above)
Comment (VikkyMaru) [reply Inception/ClientApp/src/App/Services.ts]: +

File: Inception/ClientApp/src/App/Home/HomeComponent.ts [hunk 10]
```diff
@@ -1,15 +1,31 @@
...
+    {
+
+    }
+
+    public ngOnInit(): void
+    {
+
+    }
+
+    public GetTestResultHistoryList()
```
Comment (PavelLev) [inline Inception/ClientApp/src/App/Home/HomeComponent.ts]: Rename + add return type

File: Inception/ClientApp/src/App/Home/LinkTestResultComponent.ts [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+{
+    @Input()
+    public LinkTestResult: LinkTestResult;
+
+    constructor()
+    {
+
+    }
+
+    public ngOnInit()
```
Comment (PavelLev) [inline Inception/ClientApp/src/App/Home/LinkTestResultComponent.ts]: return type

File: Inception/ClientApp/src/App/Home/TestResultHistoryComponent.ts [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+{
+    @Input()
+    public SiteTestResult: SiteTestResult;
+
+    constructor(private _testingService: TestingService, private route: ActivatedRoute)
+    {
+
+    }
+
+    public ngOnInit()
```
Comment (PavelLev) [inline Inception/ClientApp/src/App/Home/TestResultHistoryComponent.ts]: return type

File: Inception/ClientApp/src/App/Home/TestResultHistoryComponent.ts [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+        styleUrls: ["TestResultHistoryComponent.css"],
+        templateUrl: "TestResultHistoryComponent.html"
+    }
+    )
+export class TestResultHistoryComponent implements OnInit
+{
+    @Input()
+    public SiteTestResult: SiteTestResult;
+
+    constructor(private _testingService: TestingService, private route: ActivatedRoute)
```
Comment (PavelLev) [inline Inception/ClientApp/src/App/Home/TestResultHistoryComponent.ts]: rename route to _activatedRoute

File: Inception/ClientApp/src/App/Home/TestResultHistoryListComponent.ts [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+        templateUrl: "TestResultHistoryListComponent.html"
+    }
+    )
+
+export class TestResultHistoryListComponent implements OnInit
+{
+    @Input()
+    public SiteTestResult: SiteTestResult;
+
+    constructor(private _testingService: TestingService, private route: ActivatedRoute)
```
Comment (PavelLev) [inline Inception/ClientApp/src/App/Home/TestResultHistoryListComponent.ts]: rename route to _activatedRoute

File: Inception/ClientApp/src/App/Home/TestResultHistoryListComponent.ts [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+    {
+
+    }
+
+
+
+    public ngOnInit()
```
Comment (PavelLev) [inline Inception/ClientApp/src/App/Home/TestResultHistoryListComponent.ts]: return type

**Pull request merged**